Sources/PrimerSDK/Classes/Core/Primer/Primer.swift

```diff
@@ -200,6 +210,117 @@ public class Primer {
             self?.root?.dismiss(animated: true, completion: nil)
         }
     }
+    // swiftlint:disable cyclomatic_complexity
+    public func performThreeDS(paymentMethod: PaymentMethodToken, completion: @escaping (Error?) -> Void) {
+        let state: AppStateProtocol = DependencyContainer.resolve()
+        
+        guard let clientToken = state.decodedClientToken else {
+            return completion(PrimerError.vaultFetchFailed)
+        }
+        
+        let service = ThreeDSecureService()
+        service.initializeSDK { (initResult) in
+            switch initResult {
+            case .success:
+                service.verifyWarnings { (verifyResult) in
+                    switch verifyResult {
+                    case .success:
+                        service.netceteraAuth(paymentMethod: paymentMethod, protocolVersion: .v1) { (authResult) in
+                            switch authResult {
+                            case .success(let transaction):
+                                let threeDSecureAuthData = try! transaction.buildThreeDSecureAuthData()
+                                print("3DS SDK Data: \(threeDSecureAuthData)")
+                                
+                                var req = ThreeDS.BeginAuthRequest.demoAuthRequest
+                                req.device = threeDSecureAuthData
+                                req.amount = 1000
+                                
+                                service.threeDSecureBeginAuthentication(paymentMethodToken: paymentMethod,
+                                                                               threeDSecureBeginAuthRequest: req) { (res, err) in
+                                    if let err = err {
+                                        print(err)
```

Comment:
[Q]
I see a lot of `print(err)`. Do we always want to print error?

Reply:
No, I just log till the 3DS decision logic is done. Again only as long we're still testing.